dataRetriever/dataPool/proofsCache/proofsPool.go

```diff
@@ -43,22 +44,40 @@ func NewProofsPool(cleanupNonceDelta uint64, bucketSize int) *proofsPool {
 	}
 }
 
-// AddProof will add the provided proof to the pool
+// UpsertProof will add the provided proof to the pool. If there is already an existing proof,
+// it will overwrite it.
+func (pp *proofsPool) UpsertProof(
+	headerProof data.HeaderProofHandler,
+) bool {
+	if check.IfNilReflect(headerProof) {
```

Comment:
check.IfNil instead for proofs IsInterfaceNil was added for the HeaderProofHandler